### PR TITLE
[Data Logging] Custom Logger

### PR DESCRIPTION
**Pull request by dbogunowicz**
Allows the user to specify their custom leaf loggers to use in the server.
First, the user needs to specify the path to the custom logger in the yaml config:
```yaml
num_cores: 2
num_workers: 2
loggers:
    custom_logger:
        path: tests/deepsparse/loggers/helpers.py:CustomLogger
        arg1: 1
        arg2: some_string
endpoints:
    - task: question_answering
      route: /unpruned/predict
      model: zoo:nlp/question_answering/bert-base/pytorch/huggingface/squad/base-none
```
Then, the logic of `build_logger` function will attempt to instantiate the `custom_logger`.
- will look for module `tests/deepsparse/loggers/helpers.py`
- try to import `CustomLogger` from the module
- instantiate the `CustomLogger` with the specified arguments
- finally, it will also check whether `CustomLogger` inherits from our internal `BaseLogger`

### CODE REVIEW

File: src/deepsparse/server/build_logger.py [hunk 1]
```diff
@@ -139,6 +145,19 @@ def _build_function_logger(
     )
 
 
+def _build_custom_logger(logger_arguments: Dict[str, Any]) -> BaseLogger:
+    # gets the identifier from logger arguments and simultaneously
+    # remove the identifier from the arguments
```
Comment (dbogunowicz) [inline src/deepsparse/server/build_logger.py:150]:
```suggestion
    # removes the identifier from the arguments
```

File: src/deepsparse/server/helpers.py [hunk 2]
```diff
@@ -15,16 +15,34 @@
...
 
 
+def custom_logger_from_identifier(custom_logger_identifier: str) -> Type[BaseLogger]:
+    """
+    Parse the custom logger identifier in order to import a custom logger class object
+    from the user-specified python script
+
+    :param custom_logger_identifier: string in the form of
+           '<path_to_the_python_script>:<custom_logger_class_name>
+    :return: Custom logger class object
```
Comment (dbogunowicz) [inline src/deepsparse/server/helpers.py:37]:
```suggestion
    :return: custom logger class object
```

File: tests/deepsparse/loggers/helpers.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    def __init__(self, logger: BaseLogger, sleep_time: int = 1):
+        self.logger = logger
+        self.sleep_time = sleep_time
+
+    def log(self, identifier: str, value: Any, category: MetricCategories):
+        sleep(self.sleep_time)
+        self.logger.log(identifier=identifier, value=value, category=category)
+
+
+class CustomLogger(BaseLogger):
```
Comment (dbogunowicz) [inline tests/deepsparse/loggers/helpers.py:84]: This is the only contribution of this PR to the `tests/deepsparse/loggers/helpers.py` the rest of the code is carried over from https://github.com/neuralmagic/deepsparse/pull/766 (for convenience)

**Review by corey-nm — APPROVED** [review]
lgtm

File: tests/deepsparse/loggers/helpers.py (hunk 3 above)
Comment (bfineran) [reply tests/deepsparse/loggers/helpers.py:84]: In future no need to copy over, should be a simple merge

**Review by bfineran — APPROVED** [review]
LGTM

**Pull request merged**